src/containers/CreatePipelineRun/CreatePipelineRun.js

```diff
@@ -14,497 +14,427 @@ limitations under the License.
-                  : ''
-              }
-              onChange={({ selectedItem }) =>
-                this.onChange({
-                  name: 'pipeline',
-                  value: selectedItem.text
-                })
+                pipelineRef ? { id: pipelineRef, text: pipelineRef } : ''
               }
-              disabled={!!pipelineName}
-              invalid={this.state.pipeline.invalid}
-              invalidText="Pipeline cannot be empty"
+              onChange={this.handlePipelineChange}
             />
+          </FormGroup>
+          {resourceSpecs && resourceSpecs.length !== 0 && (
+            <FormGroup legendText="Pipeline Resources">
+              {resourceSpecs.map(resourceSpec => (
+                <PipelineResourcesDropdown
+                  id={`create-pipelinerun--pr-dropdown-${resourceSpec.name}`}
+                  key={`create-pipelinerun--pr-dropdown-${resourceSpec.name}`}
+                  titleText={resourceSpec.name}
+                  helperText={resourceSpec.type}
+                  type={resourceSpec.type}
+                  namespace={namespace}
+                  invalid={
+                    validationError && !this.state.resources[resourceSpec.name]
+                  }
+                  invalidText="Pipeline Resources cannot be empty"
+                  selectedItem={(() => {
+                    const value = this.state.resources[resourceSpec.name];
+                    return value ? { id: value, text: value } : '';
+                  })()}
+                  onChange={({ selectedItem: { text } }) =>
+                    this.handleResourceChange(resourceSpec.name, text)
+                  }
+                />
+              ))}
+            </FormGroup>
+          )}
+          {paramSpecs && paramSpecs.length !== 0 && (
+            <FormGroup legendText="Params">
+              {paramSpecs.map(paramSpec => (
+                <TextInput
+                  id={`create-pipelinerun--param-${paramSpec.name}`}
+                  key={`create-pipelinerun--param-${paramSpec.name}`}
+                  labelText={paramSpec.name}
+                  helperText={paramSpec.description}
+                  placeholder={paramSpec.default || paramSpec.name}
+                  invalid={
+                    validationError && !this.state.params[paramSpec.name]
+                  }
+                  invalidText="Params cannot be empty"
+                  value={this.state.params[paramSpec.name] || ''}
+                  onChange={({ target: { value } }) =>
+                    this.handleParamChange(paramSpec.name, value)
+                  }
+                />
+              ))}
+            </FormGroup>
+          )}
+          <FormGroup legendText="Optional values">
             <ServiceAccountsDropdown
-              id="dropdown-service-account"
-              namespace={namespaceValue}
+              id="create-pipelinerun--sa-dropdown"
+              helperText="Optional"
```

Comment:
optional fields should be denoted as such via their labels, e.g. 'Service Account (optional)'
https://www.carbondesignsystem.com/components/form/usage#optional-vs.-required-fields